openshift/console
# Virtual Scrolling for Resource List Views

### Description

Improves performance for huge resource lists (they don't crash) using `react-virtualized`. Includes fixes to various performance bottlenecks found due to the increased number of component mounts.

**NOTE:** The React Developer Tools Chrome extension causes choppiness with the scrolling. Use Incognito Mode or disable the extension to test.

### Video

**Pods view (~2500 pods using load-testing ServiceWorker):**
![virtual-scrolling](https://user-images.githubusercontent.com/11700385/41372127-5be76228-6f1a-11e8-93ca-249c8274bd56.gif)

### Testing

1. Run using load testing ServiceWorker: `./bin/bridge --load-test-factor 30`
2. View "Workloads"->"Pods"
3. Observe the smooth scrolling

### Future Work

- Fix text filter performance (probably debouncing)
- Address mount performance of `ResourceCog`, `LabelList`, etc
- Improve usage of `immutable` library

Addresses https://jira.coreos.com/browse/CONSOLE-467
Adopted from https://github.com/openshift/console/pull/76
